crates/router/src/core/payouts/helpers.rs

```diff
@@ -177,76 +177,115 @@ pub async fn save_payout_data_to_locker(
+                            secret
+                        })
+                        .async_lift(|inner| domain_types::encrypt_optional(inner, key))
+                        .await
+                }
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to encrypt payout method data")?
+                .map(Encryption::from)
+                .map(|e| e.into_inner())
+                .map_or(Err(errors::ApiErrorResponse::InternalServerError), |e| {
+                    Ok(hex::encode(e.peek()))
+                })?;
+                let payload = StoreLockerReq::LockerGeneric(StoreGenericReq {
+                    merchant_id: &merchant_account.merchant_id,
+                    merchant_customer_id: payout_attempt.customer_id.to_owned(),
+                    enc_data,
+                });
+                (
+                    payload,
+                    None,
+                    Some(bank.to_owned()),
+                    None,
+                    api_enums::PaymentMethodType::foreign_from(bank.to_owned()),
+                )
+            }
+            api_models::payouts::PayoutMethodData::Wallet(wallet) => {
+                let key = key_store.key.get_inner().peek();
+                let enc_data = async {
+                    serde_json::to_value(payout_method_data.to_owned())
+                        .into_report()
+                        .change_context(errors::ApiErrorResponse::InternalServerError)
+                        .attach_printable("Unable to encode payout method data")
+                        .ok()
+                        .map(|v| {
+                            let secret: Secret<String> = Secret::new(v.to_string());
+                            secret
+                        })
+                        .async_lift(|inner| domain_types::encrypt_optional(inner, key))
+                        .await
+                }
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to encrypt payout method data")?
+                .map(Encryption::from)
+                .map(|e| e.into_inner())
+                .map_or(Err(errors::ApiErrorResponse::InternalServerError), |e| {
+                    Ok(hex::encode(e.peek()))
+                })?;
+                let payload = StoreLockerReq::LockerGeneric(StoreGenericReq {
+                    merchant_id: &merchant_account.merchant_id,
+                    merchant_customer_id: payout_attempt.customer_id.to_owned(),
+                    enc_data,
+                });
+                (
+                    payload,
+                    None,
+                    None,
+                    Some(wallet.to_owned()),
+                    api_enums::PaymentMethodType::foreign_from(wallet.to_owned()),
+                )
+            }
+        };
```

Comment:
```suggestion
            _ => {
                let key = key_store.key.get_inner().peek();
                let enc_data = async {
                    serde_json::to_value(payout_method_data.to_owned())
                        .into_report()
                        .change_context(errors::ApiErrorResponse::InternalServerError)
                        .attach_printable("Unable to encode payout method data")
                        .ok()
                        .map(|v| {
                            let secret: Secret<String> = Secret::new(v.to_string());
                            secret
                        })
                        .async_lift(|inner| domain_types::encrypt_optional(inner, key))
                        .await
                }
                .await
                .change_context(errors::ApiErrorResponse::InternalServerError)
                .attach_printable("Failed to encrypt payout method data")?
                .map(Encryption::from)
                .map(|e| e.into_inner())
                .map_or(Err(errors::ApiErrorResponse::InternalServerError), |e| {
                    Ok(hex::encode(e.peek()))
                })?;
                let payload = StoreLockerReq::LockerGeneric(StoreGenericReq {
                    merchant_id: &merchant_account.merchant_id,
                    merchant_customer_id: payout_attempt.customer_id.to_owned(),
                    enc_data,
                });

                match payout_method_data {
                    api_models::payouts::PayoutMethodData::Bank(b) => {
                        (
                            payload,
                            None,
                            Some(bank.to_owned()),
                            None,
                            api_enums::PaymentMethodType::foreign_from(bank.to_owned()),
                        )
                    }
                    api_models::payouts::PayoutMethodData::Wallet(w) => {
                        (
                            payload,
                            None,
                            None,
                            Some(wallet.to_owned()),
                            api_enums::PaymentMethodType::foreign_from(wallet.to_owned()),
                        )
                    }
                }   
            }
        };
```

Encryption of data is done the same way for both banks and wallets, so we can merge the steps